src/openforms/forms/api/serializers.py

```diff
@@ -689,16 +690,20 @@ def validate(self, data: dict) -> dict:
             and action_type in LogicActionTypes.requires_component
             and not component
         ):
-            # raises validation error
-            self.fields["component"].fail("blank")
+            raise serializers.ValidationError(
+                {"component": self.fields["component"].error_messages["blank"]},
+                code="blank",
```

Comment:
did you verify that this does indeed contain the code in the error response? Afaik it gets discarded if you pass a dict to `ValidationError` instead of just an err message string.

You should be able to easily make that assertion in the tests.

Reply:
It looks like it is kept! I added the check in the tests